gen-certs.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+set -x
+set -e
+
+CACN=${CACN:-CaaSP Internal CA}
+ORG=${ORG:-SUSE Internal}
+ORGUNIT=${ORGUNIT:-CaaSP}
+CITY=${CITY:-Nuremberg}
+STATE=${STATE:-Bavaria}
+COUNTRY=${COUNTRY:-DE}
+
+dir() {
+    echo "/etc/pki"
+}
+
+work() {
+    echo "$(dir)/_work"
+}
+
+genca() {
+    [ -f $(dir)/ca.key ] && [ -f $(dir)/ca.crt ] && return
+
+    mkdir -p $(work)
+
+    # generate the CA _work key
+    openssl genrsa -out $(dir)/ca.key 4096
+
+    cat > $(work)/ca.cfg <<EOF
+[ca]
+default_ca = CA_default
+
+[CA_default]
+dir = $(dir)
+certs	= \$dir
+crl_dir	= $(work)
+database = $(work)/index.txt
+new_certs_dir	= \$dir
+
+certificate	= \$dir/ca.crt
+serial = $(work)/serial
+crlnumber	= $(work)/crlnumber
+crl = \$dir/crl.pem
+private_key	= \$dir/ca.key
+RANDFILE = \$dir/.rand
+
+default_days = 365
+default_crl_days = 30
+default_md = default
+preserve = false
+copy_extensions = copy
+
+policy          = policy_match
+
+[ policy_match ]
+countryName             = match
+stateOrProvinceName     = match
+organizationName        = match
+organizationalUnitName  = optional
+commonName              = supplied
+emailAddress            = optional
+
+[req]
+distinguished_name = req_distinguished_name
+req_extensions = v3_req
+prompt = no
+
+[req_distinguished_name]
+C = $COUNTRY
+ST = $STATE
+L = $CITY
+O = $ORG
+OU = $ORGUNIT
+CN = $CACN
+
+[v3_req]
+# Extensions to add to a certificate request
+basicConstraints = CA:TRUE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+EOF
+
+    rm -f $(work)/index.txt $(work)/index.txt.attr
+    touch $(work)/index.txt $(work)/index.txt.attr
+    echo 1000 > $(work)/serial
+    echo 1000 > $(work)/crlnumber
+
+    openssl req -batch -config $(work)/ca.cfg -sha256 -new -x509 -days 3650 -key $(dir)/ca.key -out $(dir)/ca.crt
+}
+
+gencert() {
+    genca
+
+    [ -f $(dir)/$1.key ] && [ -f $(dir)/$1.crt ] && return
+
+    # generate the server cert
+    openssl genrsa -out $(dir)/$1.key 4096
```

Comment:
I remember there was a discussion on the life of the certificates but I don't have the notes at hand. Didn't we agree on 2 years so not to have to update the certificates in a year?